code/game/objects/items/chainsaw.dm

```diff
@@ -78,6 +78,40 @@
 		var/datum/action/A = X
 		A.UpdateButtonIcon()
 
+/obj/item/chainsaw/afterattack(atom/A, mob/user, proximity)
+	if(!on)
+		return ..()
+
+	var/lignocellulose = counterlist_sum(A.has_material_type(/datum/material/wood))
+
+	if(!lignocellulose)
+		return ..()
+
+
+
+	var/dust_harvest = lignocellulose * 0.1 //you lose 10% as sawdust
+	lignocellulose *= 0.9
```

Comment:
Make the 0.1 a define, then derive 0.9 from 1 - DEFINE_NAME_HERE.

Reply:
Sure